### PR TITLE
refactor: Remove net GRPC API

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

N/A

## Description

This PR removes the GRPC API from the net package. The HTTP and CLI interfaces now include this functionality.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

`make test`

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: net/node_test.go
Comment (fredcarle) [inline net/node_test.go:1]: question: I've never seen these test being flaky. I'm not sure the tests are relevant anymore if we don't check the outcome. What do you think caused them to become flaky?

File: tests/integration/net/state/simple/replicator/with_create_update_test.go
Comment (AndrewSisley) [inline tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: todo: This is changing what the test tests, and if there is flakiness here, it should probably be handled with a `WaitForSync` action instead.

That said, where have you seen this flakiness? Do you have an issue or an error log?

File: tests/integration/p2p.go
Comment (AndrewSisley) [inline tests/integration/p2p.go:82]: praise: This is a good change, thanks - the test action should reflect the structure of the production operation it represents.  `AddP2PCollection` takes a single id, so so should the action IMO

File: tests/integration/net/state/simple/peer/subscribe/with_add_test.go
Comment (AndrewSisley) [inline tests/integration/net/state/simple/peer/subscribe/with_add_test.go:272]: question: Why have these tests been removed?

File: net/node.go
Comment (nasdf) [reply net/node.go:55]: I had to temporarily because `client.DB` now has the same method signatures. For example `SetReplicator` here and in `client.DB` are the same.

File: net/node_test.go
Comment (nasdf) [reply net/node_test.go:1]: When it failed it was logging an extra line. My guess is that it was always prone to being flaky. Now that we run the parallel test matrix it just appears more often.

File: tests/integration/net/state/simple/replicator/with_create_update_test.go
Comment (nasdf) [reply tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: The CLI tests fail with the original test setup. If the replicator syncs the document before the second node attempts to create it, then it will fail. I will swap it for a `WaitForSync` like you suggested.

Comment (AndrewSisley) [reply tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: But only the CLI tests failed/were flaky?  That suggests a problem with the production code, not the test.  

If the other mutation types are not flaky, and the CLI is, the CLI is not behaving the same way as the others.

File: tests/integration/net/state/simple/peer/subscribe/with_add_test.go
Comment (nasdf) [reply tests/integration/net/state/simple/peer/subscribe/with_add_test.go:272]: `TestP2PSubscribeAddSingleErroneousCollectionID` tests the same thing since only one subscription at a time is allowed.

Comment (AndrewSisley) [reply tests/integration/net/state/simple/peer/subscribe/with_add_test.go:272]: `TestP2PSubscribeAddSingleErroneousCollectionID` does not test that other existing collection-peer subscriptions are unaffected by the attempt to configure a bad collection ID.

File: tests/integration/net/state/simple/replicator/with_create_update_test.go
Comment (nasdf) [reply tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: I believe this is due to the CLI being the slowest implementation. If you look through the original test steps you can see where the race condition happens. The document syncs from node 0 to node 1 before node 1 has the chance to create it.

File: tests/integration/p2p.go
Comment (nasdf) [reply tests/integration/p2p.go:82]: I'm happy to consider either option as long as it matches the client API.

File: tests/integration/net/state/simple/peer/subscribe/with_add_test.go
Comment (nasdf) [reply tests/integration/net/state/simple/peer/subscribe/with_add_test.go]: My bad, I had misunderstood these tests. I restored them with one minor modification to make it pass.

File: tests/integration/net/state/simple/replicator/with_create_update_test.go
Comment (AndrewSisley) [reply tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: > I believe this is due to the CLI being the slowest implementation to test.

That does not make sense to me though, as the slowness is outside of the execution bounds in which speed matters (RE race), and if anything the CLI should be more resilient to flakiness here, as there is a larger gap between submitting the Create, and the Update.

Where x = main test thread, y = create async stuff, z = update async stuff, * = mutation type specific x stuff:
```
x * // Create doc action start
x *
x y // 
x y
x y
x y
x y // Create doc action end
x * // Update doc action start
x *
x z
x z
x z
``` 
When running CLI tests * may occupy more time, but that should only have the same effect as adding in a wait/sleep, and should reduce the risk of hitting a race condition (where y and z overlap).

From memory I also believe that the `ConfigureReplicator` action includes a wait anyway.

`TestP2PPeerReplicatorWithUpdate` also seems to run fine with the same create-config-update flow?

File: tests/integration/p2p.go
Comment (fredcarle) [reply tests/integration/p2p.go:82]: I'll let you decide which one you think creates a better user experience. 

```
defradb client p2pcollection add Users
defradb client p2pcollection add Address
```
vs
```
defradb client p2pcollection add Users,Address
```

File: tests/integration/net/state/simple/peer/subscribe/with_add_test.go
Comment (AndrewSisley) [inline tests/integration/net/state/simple/peer/subscribe/with_add_test.go:340]: todo: Either the name of this test should be changed slightly, or the test deleted (if this is tested elsewhere, which it might be).

File: tests/integration/net/state/simple/replicator/with_create_update_test.go
Comment (nasdf) [reply tests/integration/net/state/simple/replicator/with_create_update_test.go:151]: I created an issue to start tracking tests with potential race condition #1935

**Review by AndrewSisley — APPROVED** [review]
LGTM, approving now, assuming Fred's comments are addressed sensibly, as well as/including the log test stuff (maybe split that out to another PR and merge this stuff now?). Plus the one new minor todo from me :)

Thanks Keenan :)

File: tests/integration/p2p.go
Comment (nasdf) [reply tests/integration/p2p.go]: I've updated it to support multiple collections like we discussed.

**Pull request merged**
